src/L2TokenBridge.sol

```diff
@@ -85,13 +87,30 @@ contract L2TokenBridge {
         address _otherBridge,
         address _messenger
     ) {
+        _disableInitializers(); // Avoid initializing in the context of the implementation
+
         otherBridge = _otherBridge;
         messenger = CrossDomainMessengerLike(_messenger);
+    }
+
+    // --- upgradability ---
+
+    string public constant version = "1";
 
+    function initialize() initializer external {
+        __UUPSUpgradeable_init();
+
+        isOpen = 1;
         wards[msg.sender] = 1;
         emit Rely(msg.sender);
     }
 
+    function _authorizeUpgrade(address newImplementation) internal override auth {}
```

Comment:
unrelated - we might want to use the opportunity and remove the:
"// TODO: should l2Tokens allow authed burn?"

Reply:
Removed